### ISSUE TITLE
Running individual tests with `pytest` fails when not all `iris` submodules are imported inside test module

### ISSUE THREAD

**Issue by valeriupredoi** [ESMValGroup/ESMValCore#1329]
Related to #1322 (but that's closed since it served its purpose). Running tests collectively with `pytest` or `pytest -m MSSG` works fine since, as @bouweandela mentioned, imports of iris submodules are done at top level, even if not explicitly in the test module itself. But running individual test modules for debugging and development purposes, fails when `iris` submodules are not imported explictly inbthe test module. Example:

```
pytest tests/unit/cmor/test_cmor_check.py
```

dies spectacularly with multiple:
```
FAILED tests/unit/cmor/test_cmor_check.py::TestCMORCheck::test_warning_fail_on_error - AttributeError: module 'iris' has no attribute 'cube'
```

**Comment by valeriupredoi** [comment]
sorry, it took me a while to get back to this one, @bouweandela - nice scripty-script, but note that you are piping everything to dev/numm (that's stdout **and** stderr); so don't pipe and you'll see fails: like eg `tests/integration/cmor/_fixes/cmip5/test_mri_cgcm3.py` and lots others like it (lots from the fixes tests). Note that these are not failed tests, but rather errors while collecting the tests

**Comment by valeriupredoi** [comment]
for some miraculous reason, all tests now pass - without the need of explicit imports or fixes as we did in #1367 - I don't know why, but I know I am now using python=3.9.9, as opposed to 3.9.7 previously, in fact here is the pytest setup difference:

- we had fails with:
```
============================= test session starts ==============================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /home/valeriu/ESMValCore, configfile: setup.cfg
plugins: xdist-2.4.0, forked-1.3.0, mypy-0.8.1, cov-3.0.0, html-3.1.1, mock-3.6.1, flake8-1.0.7, metadata-1.11.0, env-0.6.2
```
- all fine with:
```
============================= test session starts ==============================
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/valeriu/ESMValTool, configfile: setup.cfg
plugins: html-3.1.1, metadata-1.11.0, xdist-2.5.0, forked-1.4.0, env-0.6.2, cov-3.0.0, mock-3.6.1, mypy-0.8.1, flake8-1.0.7
```
